### PR TITLE
Create nodegroups/managed-nodegroups on unowned clusters

### PR DESCRIPTION
**Pull request by Callisto13**
### Description

Most of the mechanism for creating managed and unmanaged nodegroups stays the same whether they are on an owned or an unowned cluster. This PR adds a VPC config "getter" which will fetch the required networking information from either the Cluster Stack (in the case of owned clusters), or the provided Cluster Config Spec.
To use this feature we expect users to provide their VPC spec, which I think is reasonable.

I updated some delete procedures to make sure we cover all types of nodegroups which need to die.

I also included an integration test here instead of in a separate PR.

Other random bits thrown in for free:
- Stop creating a new stack manager everywhere. Some places were creating a new stack manager, then calling a func which created another one.
- Shuffle some stuff around in ctl/create/cluster. Huge funcs inside other funcs really upsets me.
- New StackNotFoundErr type.
- Some reorganisation in actions/nodegroup create, and tidying up of how we init a new ng manager

I didn't go into refactoring here, since there are already 40+ files in this changeset. There is a lot I want to do around tests, which I will PR separately.

Still TODO:
- Tidy some unit tests (in another PR)
- Docs on running commands on non-eksctl created clusters (in another PR)

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/cfn/manager/create_tasks.go [hunk 1]
```diff
@@ -26,9 +27,9 @@ func (c *StackCollection) NewTasksToCreateClusterWithNodeGroups(nodeGroups []*ap
 	)
 
 	appendNodeGroupTasksTo := func(taskTree *tasks.TaskTree) {
-		nodeGroupTasks := c.NewUnmanagedNodeGroupTask(nodeGroups, supportsManagedNodes, false)
-
-		managedNodeGroupTasks := c.NewManagedNodeGroupTask(managedNodeGroups, false)
+		vpcImporter := vpc.NewStackConfigImporter(c.MakeClusterStackName())
```
Comment (Callisto13) [inline pkg/cfn/manager/create_tasks.go:30]: @aclevername this is the silly thing I screwed up. I initially had this as the cluster name, when it needs the stack name 🤦‍♀️

Comment (aclevername) [reply pkg/cfn/manager/create_tasks.go:30]: nice

**Review by aclevername — APPROVED** [review]
LGTM

**Comment by Callisto13** [comment]
I am running the integration action, are you okay working off this branch for the next hour?

**Comment by aclevername** [comment]
> I am running the integration action, are you okay working off this branch for the next hour?

:ok:

File: pkg/actions/nodegroup/delete.go [hunk 2]
```diff
@@ -3,25 +3,22 @@ package nodegroup
 import (
```
Comment (Callisto13) [inline pkg/actions/nodegroup/delete.go:3]: changed this file again @aclevername . 99.9999% sure it works now :p

**Comment by Callisto13** [comment]
integration tests all passing 🎉

**Comment by tobias-zeptio** [comment]
I get a error when I try to create an unmanaged nodegroup for a cluster created through cloudformation.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1e4edb8]

goroutine 1 [running]:
github.com/weaveworks/eksctl/pkg/vpc.cleanupSubnets(0xc0006a6e00)
        github.com/weaveworks/eksctl/pkg/vpc/vpc.go:529 +0x138
github.com/weaveworks/eksctl/pkg/vpc.ImportSubnetsFromSpec(0x3414528, 0xc0003320c0, 0xc0006a6e00, 0x0, 0x0)
        github.com/weaveworks/eksctl/pkg/vpc/vpc.go:492 +0x11a
github.com/weaveworks/eksctl/pkg/actions/nodegroup.loadVPCFromConfig(0x3414528, 0xc0003320c0, 0xc0006a6e00, 0x1, 0x1)
        github.com/weaveworks/eksctl/pkg/actions/nodegroup/create.go:282 +0x69
github.com/weaveworks/eksctl/pkg/actions/nodegroup.(*Manager).Create(0xc0006612c0, 0xc000010101, 0xc000a82000, 0x0, 0xc000421890, 0xc0004218c0, 0x0, 0x339e920)
        github.com/weaveworks/eksctl/pkg/actions/nodegroup/create.go:50 +0x2ef
github.com/weaveworks/eksctl/pkg/ctl/create.createNodeGroupCmd.func1(0xc0008b3b80, 0xc0008be900, 0x10101, 0x0, 0x0, 0x0, 0x0, 0xc000a1fd18, 0x1f685fd)
        github.com/weaveworks/eksctl/pkg/ctl/create/nodegroup.go:40 +0x214
github.com/weaveworks/eksctl/pkg/ctl/create.createNodeGroupCmdWithRunFunc.func1(0xc00069cf00, 0xc0009481c0, 0x0, 0xe, 0x0, 0x0)
        github.com/weaveworks/eksctl/pkg/ctl/create/nodegroup.go:62 +0x10a
github.com/spf13/cobra.(*Command).execute(0xc00069cf00, 0xc0009480e0, 0xe, 0xe, 0xc00069cf00, 0xc0009480e0)
        github.com/spf13/cobra@v1.1.3/command.go:852 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc00069c280, 0xc000384670, 0x1, 0xc000a1ff70)
        github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
        github.com/weaveworks/eksctl/cmd/eksctl/main.go:76 +0x41b
```

This is the command I run:

```
eksctl create nodegroup \
    --cluster husqiot-eks-dev \
    --node-type t3a.small \
    --node-ami auto \
    --nodes 1 \
    --nodes-min 1 \
    --nodes-max 2 \
    --max-pods-per-node 100
```

And this is the cloudformation:

```
  ServiceRole:
    Type: AWS::IAM::Role
    Properties:
      RoleName: "eks-servicerole"
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - eks.amazonaws.com
            Action:
              - sts:AssumeRole
      ManagedPolicyArns:
        - arn:aws:iam::aws:policy/AmazonEKSClusterPolicy

  EKSCluster:
    Type: "AWS::EKS::Cluster"
    Properties:
      Name: !Sub "husqiot-eks-dev"
      Version: "1.19"
      RoleArn: !GetAtt ServiceRole.Arn
      ResourcesVpcConfig:
        SubnetIds:
          Fn::Split:
            - ","
            - Fn::Join:
                - ","
                - - Fn::ImportValue: !Sub "${NetworkingStackName}-PublicSubnets"
                  - Fn::ImportValue: !Sub "${NetworkingStackName}-PrivateSubnets"
```

**Comment by Callisto13** [comment]
@tobias-zeptio could you open a new issue please?

**Comment by tobias-zeptio** [comment]
@Callisto13 Sorry about that, here it is. #3484

**Pull request merged**